sys-libs/llvm-libunwind/files/Makefile

```diff
@@ -0,0 +1,43 @@
+ASFLAGS+=-Wa,--noexecstack
```

Comment:
Please try. LLVM upstream is quite likely to accept a good patch to support this.